src/main/windows/editor.js

```diff
@@ -17,72 +16,108 @@ class EditorWindow extends BaseWindow {
 
     const mainWindowState = windowStateKeeper({
       defaultWidth: 1200,
       defaultHeight: 800
     })
 
     const { x, y, width, height } = ensureWindowPosition(mainWindowState)
-    const winOptions = Object.assign({ x, y, width, height }, defaultWinOptions, options)
+    const winOptions = Object.assign({ x, y, width, height }, editorWinOptions, options)
     if (isLinux) {
       winOptions.icon = path.join(__static, 'logo-96px.png')
     }
 
     // Enable native or custom/frameless window and titlebar
-    const { titleBarStyle } = preferences.getAll()
+    const {
+      titleBarStyle,
+      theme,
+      sideBarVisibility,
+      tabBarVisibility,
+      sourceCodeModeEnabled
+    } = preferences.getAll()
     if (!isOsx) {
       winOptions.titleBarStyle = 'default'
       if (titleBarStyle === 'native') {
         winOptions.frame = true
       }
     }
 
+    winOptions.backgroundColor = this._getPreferredBackgroundColor(theme)
+
     let win = this.browserWindow = new BrowserWindow(winOptions)
     this.id = win.id
 
     // Create a menu for the current window
-    appMenu.addEditorMenu(win)
-
-    win.once('ready-to-show', async () => {
-      mainWindowState.manage(win)
-      win.show()
-
-      this.emit('window-ready-to-show')
-
-      if (pathname && isMarkdownFile(pathname)) {
-        // Open single markdown file
-        appMenu.addRecentlyUsedDocument(pathname)
-        this._openFile(pathname)
-      } else if (pathname && isDirectory(pathname)) {
-        // Open directory / folder
-        appMenu.addRecentlyUsedDocument(pathname)
-        this.openFolder(pathname)
-      } else {
-        // Open a blank window
-        const lineEnding = preferences.getPreferedEOL()
-        win.webContents.send('mt::bootstrap-blank-window', {
-          lineEnding,
-          markdown
-        })
-        appMenu.updateLineEndingMenu(lineEnding)
-      }
+    appMenu.addEditorMenu(win, { sourceCodeModeEnabled })
+
+    win.webContents.once('did-finish-load', () => {
+      this.lifecycle = WindowLifecycle.READY
+      this.emit('window-ready')
+
+      // Restore and focus window
+      this.bringToFront()
+
+      const lineEnding = preferences.getPreferedEOL()
+      appMenu.updateLineEndingMenu(lineEnding)
+
+      win.webContents.send('mt::bootstrap-editor', {
+        addBlankTab,
+        markdownList: this._markdownToOpen,
+        lineEnding,
+        sideBarVisibility,
+        tabBarVisibility,
+        sourceCodeModeEnabled
+      })
+
+      this._doOpenFilesToOpen()
+      this._markdownToOpen.length = 0
+    })
+
+    win.webContents.once('did-fail-load', (event, errorCode, errorDescription) => {
+      log.error(`The window failed to load or was cancelled: ${errorCode}; ${errorDescription}`)
+    })
+
+    win.webContents.once('crashed', (event, killed) => {
+      const msg = `The renderer process has crashed unexpected or is killed (${killed}).`
```

Comment:
Do we need to report the error on GitHub issue?

Reply:
We run Electron's `crashReporter` in main and renderer processes that dumps a native crash but I think that this case should almost never happen. `crashed` is only emitted for native crashes and when the process is killed.